service/src/main/java/uk/gov/hmcts/reform/fpl/service/DocmosisCoverDocumentsService.java

```diff
@@ -17,22 +16,25 @@
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class DocmosisCoverDocumentsService {
     private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+    private final ObjectMapper mapper;
 
     public DocmosisDocument createCoverDocuments(String familyManCaseNumber,
                                                  Long caseNumber,
                                                  Representative representative) {
-        Map<String, Object> templateData = buildCoverDocumentsData(familyManCaseNumber, caseNumber, representative);
-        return docmosisDocumentGeneratorService.generateDocmosisDocument(templateData, COVER_DOCS);
+        DocmosisCoverDocument coverDocumentData = buildCoverDocumentsData(familyManCaseNumber,
+                                                                          caseNumber,
+                                                                          representative);
+        return docmosisDocumentGeneratorService.generateDocmosisDocument(coverDocumentData.toMap(mapper), COVER_DOCS);
```

Comment:
I guess we don't need to worry too much here, so I am fine if you don't want to do it... but I have overloaded the generateDocmosisDocument method as 

`public DocmosisDocument generatedDocmosisDocument(AbstractDocmosisData templateData, DocmosisTemplates template) {
        return generateDocmosisDocument(templateData.toMap(mapper), template);
    }
`

might make sense to do the same here?